src/coprocessor/codec/data_type/vector.rs

```diff
@@ -496,31 +473,26 @@ macro_rules! impl_as_slice {
             }
         }
 
-        impl AsMut<[Option<$ty>]> for VectorValue {
-            #[inline]
-            fn as_mut(&mut self) -> &mut [Option<$ty>] {
-                self.$mut_name()
-            }
-        }
+        // `AsMut` is not implemented intentionally.
     };
 }
 
-impl_as_slice! { Int, as_int_slice, as_mut_int_slice }
-impl_as_slice! { Real, as_real_slice, as_mut_real_slice }
-impl_as_slice! { Decimal, as_decimal_slice, as_mut_decimal_slice }
-impl_as_slice! { Bytes, as_bytes_slice, as_mut_bytes_slice }
-impl_as_slice! { DateTime, as_date_time_slice, as_mut_date_time_slice }
-impl_as_slice! { Duration, as_duration_slice, as_mut_duration_slice }
-impl_as_slice! { Json, as_json_slice, as_mut_json_slice }
+impl_as_slice! { Int, as_int_slice }
+impl_as_slice! { Real, as_real_slice }
+impl_as_slice! { Decimal, as_decimal_slice }
+impl_as_slice! { Bytes, as_bytes_slice }
+impl_as_slice! { DateTime, as_date_time_slice }
+impl_as_slice! { Duration, as_duration_slice }
+impl_as_slice! { Json, as_json_slice }
 
 macro_rules! impl_push {
     ($ty:tt, $name:ident) => {
         impl VectorValue {
-            /// Pushes a value into the column in specific type.
+            /// Pushes a value in specified concrete type into current column.
             ///
             /// # Panics
             ///
-            /// Panics if current column does not match the type.
+            /// Panics if current column is does not match the type.
```

Comment:
```suggestion
            /// Panics if the current column does not match the type.
```

Reply:
There is already a `the` referring to the type, do we need another `the`? (sorry for my poor english)

Reply:
I think whether `a` or `the` is ok here. But we need an article actually.

Reply:
Fixed in PR https://github.com/tikv/tikv/pull/4215 : https://github.com/tikv/tikv/pull/4215/commits/d91b4105f1a154b62a164d19f1475a42fc53924b